tools/fp20compiler/ts1.0_inst_list.cpp

```diff
@@ -219,6 +219,61 @@ void InstList::Invoke()
     printf(");\n");
     printf("p += 2;\n");
 
+    // Process texture stage mode arguments
+    for (i=0; i<size; i++) {
+        switch(list[i].opcode.word) {
+        case TSP_OFFSET_2D_SCALE: {
+            float matrix[2*2] = {
+                list[i].args[1], list[i].args[2],
+                list[i].args[3], list[i].args[4]
+            };
+            float offset = list[i].args[5];
+            float scale = list[i].args[6];
+
+            printf("pb_push(p++, NV097_SET_TEXTURE_SET_BUMP_ENV_MAT + %d * 64, 6);\n", i);
+            printf("*p++ = 0x%08x; /* NV097_SET_TEXTURE_SET_BUMP_ENV_MAT m[0] */\n", *(unsigned int*)&matrix[0]);

```

Comment:
nit: Consider a `static` helper function like `FloatAsInteger` (or a better name). It should also:
```c
assert(sizeof(float) == 4);
assert(sizeof(unsigned int) == 4);
```

alternatively (this needs more work as we have to change a couple of things), we could potentially rewrite:

```c
float args[TSP_MAX_ARGS];
```
to be:
```c
union Arg {
    float value;
    unsigned int value_raw;
};
Arg args[TSP_MAX_ARGS];
```

We probably still want that `sizeof` assert somewhere (potentially in `main()` as a self-check):
```c
assert(sizeof(Arg.value) == 4);
assert(sizeof(Arg.value_raw) == 4);
```